retworkx/__init__.py

```diff
@@ -1321,6 +1323,13 @@ def networkx_converter(graph):
             for x in graph.edges(data=True)
         ]
     )
+
+    if keep_attributes:
+        for node, node_index in node_indices.items():
+            attributes = graph.nodes[node]
+            attributes["__networkx_name__"] = node
```

Comment:
I'm wondering if networkx ``__network_name__`` is the best choice of key here. This will be the data payload for the node in the networkx graph right? It may not be a string or even have a `__str__()` or `__repr__()` method defined. IIRC the only requirement for a node in a networkx graph is that is hashable. Maybe something like ``__networkx_node__``?

Reply:
Fair point. I briefly considered trying to override the `__repr__` of the dict (or something like that), but decided against it.